lib/view_models/page_view_model/join_organization_viewModel.dart

```diff
@@ -0,0 +1,358 @@
+      toastDuration: const Duration(seconds: 3),
+    );
+  }
+
+  void searchOrgName(String orgName) {
+    //it is the search bar to search the organization
+    _filteredOrgInfo.clear();
+    if (orgName.isNotEmpty) {
+      for (int i = 0; i < organizationInfo.length; i++) {
+        final String name = organizationInfo[i]['name'].toString();
+        if (name.toLowerCase().contains(orgName.toLowerCase())) {
+          _filteredOrgInfo.add(organizationInfo[i]);
+          notifyListeners();
+        }
+      }
+    } else {
+      _filteredOrgInfo.add(organizationInfo);
+      notifyListeners();
+    }
+  }
+
+  // Function for getting the current user id.
+  Future<void> getCurrentUserId() async {
+    _currentUserId = await _pref.getUserId();
+    notifyListeners();
+  }
+
+  Future fetchOrg() async {
+    getCurrentUserId();
+    //function to fetch the org from the server
+    final GraphQLClient _client = graphQLConfiguration.authClient();
+
+    final QueryResult result = await _client
+        .query(QueryOptions(documentNode: gql(_query.fetchOrganizations)));
+
+    // Get the details of the current user.
+    final QueryResult userDetailsResult = await _client.query(QueryOptions(
+        documentNode: gql(_query.fetchUserInfo),
+        variables: {'id': _currentUserId}));
+
+    if (result.hasException || userDetailsResult.hasException) {
+      print(result.exception);
+      showError(result.exception.toString());
+    } else if (!result.hasException &&
+        !disposed &&
+        !userDetailsResult.hasException) {
+      _organizationInfo = result.data['organizations'] as List;
+      _organizationInfo = result.data['organizations'] as List;
+
+      // Get the details of joined organizations.
+      joinedOrganizations =
+          userDetailsResult.data['users'][0]['joinedOrganizations'] as List;
+
+      // Get the id's of joined organizations.
+      joinedOrganizations.forEach((element) {
+        joinedOrganizationsIds.add(element['_id']);
+      });
+
+      // Filtering out organizations that are created by current user.
+      _organizationInfo = organizationInfo
+          .where((element) => element['admins'][0]['_id'] != _currentUserId)
+          .toList();
+
+      // Filtering out organizations that are already joined by user.
+      joinedOrganizationsIds.forEach((e) {
+        print(e);
+        _organizationInfo =
+            organizationInfo.where((element) => element['_id'] != e).toList();
+      });
+      notifyListeners();
+    }
+  }
+
+  Future joinPrivateOrg(BuildContext context) async {
+    //function called if the person wants to enter a private organization
+    final GraphQLClient _client = graphQLConfiguration.authClient();
+
+    final QueryResult result = await _client.mutate(MutationOptions(
+        documentNode: gql(_query.sendMembershipRequest(_itemIndex))));
+
+    if (result.hasException &&
+        result.exception.toString().substring(16) == accessTokenException) {
```

Comment:
what does 16 mean in this context. Can you name a const or static variable to represent this?

Reply:
Please put lines 283 & 284 in a function **retrieveExceptionType** and let it return the type of exception which you can check and handle appropriately.  We can then reuse.
`
if result.hasException(){
   exceptionType = retrieveExceptionType(result);
}

if (exceptionType == 'accessTokenException'){
  _authController.getNewToken();
} else {
//What happens if there is some other exception - do we need to act on that?
}